src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/service/YFlowDeleteService.java

```diff
@@ -0,0 +1,159 @@
+import org.openkilda.wfm.topology.flowhs.fsm.yflow.delete.YFlowDeleteContext;
+import org.openkilda.wfm.topology.flowhs.fsm.yflow.delete.YFlowDeleteFsm;
+import org.openkilda.wfm.topology.flowhs.fsm.yflow.delete.YFlowDeleteFsm.Event;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class YFlowDeleteService
+        extends FsmBasedFlowProcessingService<YFlowDeleteFsm, Event, YFlowDeleteContext, YFlowDeleteHubCarrier> {
+    private final YFlowDeleteFsm.Factory fsmFactory;
+
+    public YFlowDeleteService(YFlowDeleteHubCarrier carrier, PersistenceManager persistenceManager,
+                              FlowResourcesManager flowResourcesManager,
+                              FlowDeleteService flowDeleteService, int speakerCommandRetriesLimit) {
+        super(new FsmExecutor<>(Event.NEXT), carrier, persistenceManager);
+        fsmFactory = new YFlowDeleteFsm.Factory(carrier, persistenceManager,
+                flowResourcesManager, flowDeleteService, speakerCommandRetriesLimit);
+
+        addFlowDeleteEventListener(flowDeleteService);
+    }
+
+    private void addFlowDeleteEventListener(FlowDeleteService flowDeleteService) {
+        flowDeleteService.addEventListener(new FlowDeleteEventListener() {
+            @Override
+            public void onCompleted(String flowId) {
+                YFlowDeleteContext context = YFlowDeleteContext.builder().subFlowId(flowId).build();
+                fsms.forEach((k, fsm) -> {
+                    if (fsm.getPendingSubFlows().contains(flowId)) {
+                        fsmExecutor.fire(fsm, Event.SUB_FLOW_REMOVED, context);
+                    }
+                });
+            }
+
+            @Override
+            public void onFailed(String flowId, String errorReason, ErrorType errorType) {
+                YFlowDeleteContext context = YFlowDeleteContext.builder()
+                        .subFlowId(flowId)
+                        .error(errorReason)
+                        .errorType(errorType)
+                        .build();
+                fsms.forEach((k, fsm) -> {
+                    if (fsm.getPendingSubFlows().contains(flowId)) {
+                        fsmExecutor.fire(fsm, Event.SUB_FLOW_FAILED, context);
+                    }
+                });
+            }
+        });
+    }
+
+    /**
+     * Handles request for y-flow deletion.
+     *
+     * @param key command identifier.
+     * @param request request data.
+     */
+    public void handleRequest(String key, CommandContext commandContext, YFlowDeleteRequest request)
+            throws DuplicateKeyException {
+        log.debug("Handling y-flow delete request with key {} and flow ID {}", key, request.getYFlowId());
+
+        if (fsms.containsKey(key)) {
+            throw new DuplicateKeyException(key, "There's another active FSM with the same key");
+        }
+
+        String eventKey = commandContext.getCorrelationId();
+        if (flowEventRepository.existsByTaskId(eventKey)) {
```

Comment:
Not required any more (because of #4579)

Reply:
Removed after rebase.